ui/src/components/DetectionsTable.tsx

```diff
@@ -50,6 +62,17 @@ export default function DetectionsTable({
     },
   });
 
+  const { notificationsForCandidate: notifications } =
+    useNotificationsForCandidateQuery({ candidateId: candidate.id }).data ?? {};
+
+  const cancelNotification = useCancelNotificationMutation({
+    onSuccess: () => {
+      queryClient.invalidateQueries(
+        useNotificationsForCandidateQuery.getKey({ candidateId: candidate.id }),
```

Comment:
I haven't actually tried out this part of the UI, it might need to be revisited based on my comment https://github.com/orcasound/orcasite/pull/289#discussion_r1436662525 in the other PR

Reply:
Just tried it locally, seems to refetch the query just fine

Reply:
Ok swapped this to using `refetch`

Reply:
I played around with this a bit more, I think the ideal way to do it is using [setQueryData](https://tanstack.com/query/v4/docs/react/guides/updates-from-mutation-responses). It would roughly look something like this:

```typescript
const cancelNotification = useCancelNotificationMutation({
    onSuccess: (data) => {
      queryClient.setQueryData<NotificationsForCandidateQuery>(
        useNotificationsForCandidateQuery.getKey({ candidateId: candidate.id }),
        (oldData) => {
          if (!oldData) return;
          return {
            ...oldData,
            notificationsForCandidate: oldData.notificationsForCandidate.map(
              (notification) =>
                notification.id === data.cancelNotification?.result?.id
                  ? data.cancelNotification?.result
                  : notification,
            ),
          };
        },
      );
    },
  });
```

But it's kind of painful to do and I don't think this pattern is worth it, we can just refetch for now and revisit later if it's causing problems. Might also be worth checking out https://github.com/klis87/normy as a way to avoid the manual updating.